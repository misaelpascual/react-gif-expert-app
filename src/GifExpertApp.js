import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Cats'])

  return (
    <>
      <section className='section'>
        <h2>
          GifExpertApp
        </h2>
        <AddCategory
          setCategories={ setCategories }
        />
        <hr />

        <div>
          { categories.map((category, index) => (
              <GifGrid
                key={index}
                category={category}
              />
            ))
          }
        </div>
      </section>
    </>
  )
}

