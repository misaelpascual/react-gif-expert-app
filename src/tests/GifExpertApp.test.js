import { shallow } from 'enzyme'
import { GifExpertApp } from '../GifExpertApp'

describe('Pruebas en <GifExpertApp />', () => {

  test('Debe mostrar <GifExpertApp /> correctamente', () => {

    const wrapper = shallow(<GifExpertApp />)

    expect(wrapper).toMatchSnapshot()

  })

  test('Debe de mostrar una lista de categorias', () => {

    const categories = ['Games']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)

  })

})
