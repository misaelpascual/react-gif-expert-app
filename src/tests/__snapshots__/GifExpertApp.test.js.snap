// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <section
    className="section"
  >
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <div>
      <GifGrid
        category="Games"
        key="0"
      />
    </div>
  </section>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <section
    className="section"
  >
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <div />
  </section>
</Fragment>
`;
